src/DotLiquid/Template.cs

```diff
@@ -195,16 +199,19 @@ public static void RegisterValueTypeTransformer(Type type, Func<object, object>
                 return transformer;
 
             // Check for interfaces
-            var interfaces = type.GetTypeInfo().ImplementedInterfaces;
-            foreach (var interfaceType in interfaces)
+            return ValueTypeTransformerCache.GetOrAdd(type, (key) =>
             {
-                if (ValueTypeTransformers.TryGetValue(interfaceType, out transformer))
-                    return transformer;
-                if (interfaceType.GetTypeInfo().IsGenericType && ValueTypeTransformers.TryGetValue(
-                    interfaceType.GetGenericTypeDefinition(), out transformer))
-                    return transformer;
-            }
-            return null;
+                var interfaces = type.GetTypeInfo().ImplementedInterfaces;
+                foreach (var interfaceType in interfaces)
+                {
+                    if (ValueTypeTransformers.TryGetValue(interfaceType, out transformer))
+                        return transformer;
+                    if (interfaceType.GetTypeInfo().IsGenericType && ValueTypeTransformers.TryGetValue(
+                        interfaceType.GetGenericTypeDefinition(), out transformer))

```

Comment:
nit, formatting: consider starting each line with a condition. If you need to multi-line parameters, consider starting each line with a parameter and putting each parameter on its own line